docs/source/llms/openai/notebooks/openai-embeddings-generation.ipynb

```diff
@@ -0,0 +1,513 @@
+    "\n",
+    "### In This Tutorial:\n",
+    "\n",
+    "- **Embedding Endpoint Configuration**: Setting up and utilizing OpenAI's embedding endpoints in MLflow.\n",
+    "- **Real-world Application**: Practical examples showcasing the use of OpenAI embeddings in machine learning projects.\n",
+    "- **Efficiency and Precision Enhancements**: Techniques for improving model performance using OpenAI embeddings.\n",
+    "\n",
+    "By the end of this tutorial, you'll have a thorough understanding of how to integrate and leverage OpenAI embeddings in your MLflow projects, harnessing the power of advanced NLP techniques.\n",
+    "\n",
+    "### Required packages:\n",
+    "\n",
+    "In order to run this tutorial, you will need to install ``beautifulsoup4`` from PyPI.\n",
+    "\n",
+    "Let's dive into the world of embeddings and explore their transformative impact on machine learning models!"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import warnings\n",
+    "\n",
+    "# Disable a few less-than-useful UserWarnings from setuptools and pydantic\n",
+    "warnings.filterwarnings(\"ignore\", category=UserWarning)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "import requests\n",
+    "\n",
+    "from bs4 import BeautifulSoup\n",
+    "import numpy as np\n",
+    "import openai\n",
+    "\n",
+    "import mlflow\n",
+    "from mlflow.models.signature import ModelSignature\n",
+    "from mlflow.types.schema import ColSpec, ParamSchema, ParamSpec, Schema, TensorSpec\n",
+    "\n",
+    "assert \"OPENAI_API_KEY\" in os.environ, \" OPENAI_API_KEY environment variable must be set\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Integrating OpenAI Model with MLflow for Document Similarity\n",
+    "\n",
+    "In this tutorial segment, we demonstrate the process of setting up and utilizing an OpenAI embedding model within MLflow for document similarity tasks.\n",
+    "\n",
+    "#### Key Steps:\n",
+    "\n",
+    "1. **Set MLflow Experiment**: We begin by setting the experiment context in MLflow, specifically for document similarity, using `mlflow.set_experiment(\"Documentation Similarity\")`.\n",
+    "\n",
+    "2. **Logging the Model in MLflow**: We initiate an MLflow run and log the OpenAI model. The model in focus is \"text-embedding-ada-002\", chosen for its robust embedding capabilities. During this step, we detail the model, the embedding task, input/output schemas, and parameters like batch size.\n",
```

Comment:
nit: Probably it's worth mention that we don't load/store the model weight locally, it just logs metadata and environment. This may leads to mis-conception that it stores a local model and thus free.

Reply:
Added words to this effect :) good call